docs/www/tls-kubernetes.md

```diff
@@ -0,0 +1,375 @@
+```
+
+Typically, a client runs in a separate pod. If that pod is within the same Kubernetes cluster and namespace, the client pod can directly mount the certificate secrets:
+
+```
+spec:
+  template:
+    spec:
+      volumes:
+        - name: _tlscert_
+          secret:
+            secretName: <cluster>-redpanda
+...
+ containers:
+        - name: client
+          volumeMounts:
+            - mountPath: /etc/tls/certs
+              name: _tlscert_
+```
+
+You can find a working example with mTLS here: [https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-with-client-auth](https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-with-client-auth)
+
+If the client pod runs in a separate namespace within the same Kubernetes cluster, the secret needs to be copied over.
+
+Similarly, if the client is on a separate network, but accessible through an external IP address or name (see [https://vectorized.io/docs/kubernetes-connectivity](https://vectorized.io/docs/kubernetes-connectivity)) the secrets need to become available by copying over the resource or exporting the CA `ca.crt` and key-pair `tls.crt` and `tls.key` as files. For example, to retrieve tls.crt:
+
+```
+kubectl get secret <cluster>-user-client -o go-template='{{index .data "ca.crt"}}' | base64 -d - > tls.crt
+```
+
+Alternatively, you can retrieve the whole resource:
+
+```
+kubectl get secret <cluster>-_user-client_ --namespace=default -o yaml
+```
+
+Once the secrets are available as files, it is possible to use `rpk` to connect with the brokers. Here we consider external brokers, but they could be replaced with internal addresses if accessible:
+
+For convenience, let's create a config file and store our values there. You can pass this as arguments to your `rpk` commands but they can become very long.
+
+Create `rpk-config.yaml`,
+
+```
+rpk:
+  kafka_api:
+    brokers:
+      - 0.external-domain.:<nodeport>
+      - 1.external-domain.:<nodeport>
+      - 2.external-domain.:<nodeport>
+    tls:
+      key_file: tls.key
+      cert_file: tls.crt
+      truststore_file: ca.crt
+```
+
+Pass your saved `rpk-config.yaml` as a flag to `rpk` in order to talk to Redpanda:
+
+```
+rpk cluster info --config rpk-config.yaml
+```
+
+You can find more examples in our GitHub folder:
+* [https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-given-cert-secret](https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-given-cert-secret)
+* [https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-given-issuer-with-client-auth](https://github.com/vectorizedio/redpanda/tree/dev/src/go/k8s/tests/e2e/create-topic-given-issuer-with-client-auth)
+
+## Disabling TLS
+
+If you have an use case where you need to get from an enabled TLS configuration to a disabled one, all you'll have to do is to remove the flag from the `yaml` file and restart your cluster.
+
```

Comment:
I'd make this all one paragraph.

Reply:
You mean like removing the header?

Reply:
The point is to stress that we do not recommend this to be done. @vsaraswat agree?

Reply:
My first thought was that this shouldn't be in the doc at all. In general, we should not be telling people how to make things less secure. Unless there's some reason for it, I think we could take out this section altogether. Support can do this kind of thing with the customer, but I don't know that this should be in the product doc. If we do want to keep it, maybe move the "we don't recommend doing this" to the top of the section and put the other paragraph below it?

Reply:
Let's remove this from the doc for now.  If we start getting a lot of requests in community slack on how to disable TLS, we can add back and have clearly labeled as "not recommended."

Reply:
Removed.